src/value.rs

```diff
@@ -113,6 +120,99 @@ impl<'gc> Value<'gc> {
         }
     }
 
+    // A small helper function to handle division-like zero handling
+    fn safe_div<T: PartialEq + ToPrimitive + Zero>(
+        lhs: T,
+        rhs: T,
+        f: &Fn(T, T) -> Value<'gc>,
+    ) -> Value<'gc> {
+        match (lhs, rhs) {
+            // Seems that all nans are negative in lua
+            (ref a, ref b) if a.is_zero() && b.is_zero() => Value::Number(-f64::NAN),
+            (ref a, ref b) if b.is_zero() => {
+                Value::Number(copysign(f64::INFINITY, a.to_f64().unwrap()))
+            }
+            (a, b) => f(a, b),
+        }
+    }
+
+    // This operation always returns a Number, even when called by int arguments
```

Comment:
The documentation on all the pub functions should be doc comments (though none of the doc comments in the whole project right now are really very *good* documentation, they're intended as at least placeholder doc comments.

Reply:
I will look up proper doc formatting and update this then.

Reply:
It doesn't have to be PROPER, just add three '///' is fine for now